api/src/main/java/org/apache/iceberg/expressions/Binder.java

```diff
@@ -140,6 +140,11 @@ public Expression or(Expression leftResult, Expression rightResult) {
       throw new IllegalStateException("Found already bound predicate: " + pred);
     }
 
+    @Override
+    public <T> Expression predicate(BoundSetPredicate<T> pred) {
```

Comment:
Let's remove this method. Right now, all of the implementations of it are identical to `predicate(BoundPredicate<T>)`. I'm not sure that we will need this.